safekeeper/src/timeline_eviction.rs

```diff
@@ -73,6 +73,10 @@ impl Manager {
     #[instrument(name = "evict_timeline", skip_all)]
     pub(crate) async fn evict_timeline(&mut self) -> bool {
         assert!(!self.is_offloaded);
+        let Some(storage) = self.wal_backup.get_storage() else {
+            warn!("no remote storage configured, skipping uneviction");
```

Comment:
This should probably return an error. Ditto unevict.

Reply:
What we really want is to disallow getting here at all without remote_storage configured.
I.e., disallow `--enable-offload` if `--remote-storage` is not set.

That's more surgery than I want to do in this PR though / have time for in the bounds of fixing
- https://github.com/neondatabase/neon/issues/11501

So, not doing.